earthaccess/dmrpp_zarr.py

```diff
@@ -139,9 +159,36 @@ def open_virtual_mfdataset(
             parser=DMRPPParser(group=group),
             preprocess=preprocess,
             parallel=parallel,
+            combine="nested",
             **xr_combine_nested_kwargs,
         )
 
+    if load:
+        if reference_dir is None:
+            ref_dir = Path(tempfile.gettempdir())
+        else:
+            ref_dir = Path(reference_dir)
+            ref_dir.mkdir(exist_ok=True, parents=True)  # type: ignore
+
+        group_name = group
```

Comment:
One of the things we briefly discussed in our call today was special characters in group names/paths. There are a few possible things I'd be worried about:

* A singly nested group, but specified with a leading slash (e.g., `/geolocation` versus `geolocation`).
* Multiple nested groups (e.g., `/group_one/group_two`).
* Group names with spaces in them.

[I've played around with this sort of thing before](https://github.com/nasa/earthdata-varinfo/blob/main/varinfo/umm_var.py#L405), for things like CMR native IDs. There I just did something equivalent to:

```
group_name = group.replace('/', '_').replace(' ', '_').lstrip('_')
```

Might be worth doing similar sanitising on a user-specified group?

Reply:
we definitely need this!

Reply:
can you suggest the commit so I can apply it?  first let me fix mypy as usual...